extensions/color-casket/src/utilities.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+
+export default function asyncEffect<T>(action: Promise<T>, callback: (state: T) => void) {
+  useEffect(() => {
+    let cancel = false;
+    action.then((state) => {
+      if (!cancel) {
+        callback(state);
+      }
+    });
+
+    return () => {
+      cancel = true;
+    };
+  }, []);
+}
+
+export function debounce<A = unknown, R = void>(fn: (args: A) => R, ms: number): [(args: A) => Promise<R>, () => void] {
+  let timer: NodeJS.Timeout;
+
+  const debounce = (args: A): Promise<R> =>
+    new Promise((resolve) => {
+      if (timer) {
+        clearTimeout(timer);
+      }
```

Comment:
Calling `clearTimeout` to an invalid (e.g: undefined) id does nothing, so this can be a tiny bit shorter.
```suggestion
      clearTimeout(timer);
```